### ISSUE TITLE
It's unclear what types `Resolve(...)` can return.

### ISSUE THREAD

**Issue by Stebalien** [ipfs/go-ipld-format#25]
For example, let's say that I'm expecting a number. Should I try casting to a uint64?, float?, etc. We should either provide "canonical" types aliases:

```go
type Integer uint64
type String string
// ...
```

Or return a custom type:

```go
type Value interface{}

func (v Value) Int() uint64, error {}
func (v Value) String() string, error {}
func (v Value) Nil() error {}
```

**Comment by whyrusleeping** [comment]
We can't know what it returns though. It can return literally anything.

**Comment by Stebalien** [comment]
It can't, e.g., return a `peer.ID`, can it? That is, I assume it can only return primitives. The problem is that it's unclear what primitives it can return. See the example in the issue body; CBOR, e.g., encodes integers as varints but nothing indicates if `go-ipld-cbor` can return `int`, `int32`, `int64`, `BigInt`, `float32`, or all of the above.

Basically, we need consistent set of primitive types.

**Comment by whyrusleeping** [comment]
No, it could totally return a peer.ID, or a ProtoNode, or an ethereum
transaction, or anything else really

**Comment by Kubuxu** [comment]
CBOR also can encode either floating point or normal integers.
